### PR TITLE
UpdateableModel: Minor optimizations

### PR DESCRIPTION
**Pull request by frankosterfeld**
Avoid some copies and detaching, include QModelIndex header.

### CODE REVIEW

File: qt/model_view/updateableModel/UpdateableModel.h [hunk 1]
```diff
@@ -29,6 +29,7 @@
 #ifndef UPDATEABLEMODEL_H
 #define UPDATEABLEMODEL_H
 
+#include <QModelIndex>
```
Comment (AndreSomers) [inline qt/model_view/updateableModel/UpdateableModel.h:32]: Why is this needed?

**Review by AndreSomers — COMMENTED** [review]
Fine in general, just wondering what the additional #include is needed for?

**Comment by frankosterfeld** [comment]
Moving to KDAB gerrit
